http-server-netty/src/main/java/io/micronaut/http/server/netty/NioEventLoopGroupFactory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.netty;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadFactory;
+
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Prototype;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Internal;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+
+/**
+ * Factory for NioEventLoopGroup.
+ * @author croudet
+ */
+@Factory
+@Internal
+@Requires(missingBeans = {EpollEventLoopGroupFactory.class, KQueueEventLoopGroupFactory.class})
+class NioEventLoopGroupFactory implements EventLoopGroupFactory {
```

Comment:
This class should either not be a factory, or not be injected directly. Factories are responsible for creating beans and should not be used directly.

Reply:
Class declared as Singleton.